File: c/apiroximate.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <complex.h>

double complex random_point() {
    double x = (double)rand() / RAND_MAX;
    double y = (double)rand() / RAND_MAX;
    return x + y * I;
}

double complex *random_points(unsigned int count) {
    double complex *result = malloc(count * sizeof(double complex));
    for (int i = 0; i < count; i++) {
        result[i] = random_point();
    }
    return result;
}

double approximate(unsigned int count) {
    unsigned int in_circle = 0;
    double complex *points = random_points(count);
    for (int i = 0; i < count; i++) {
        if (cabs(points[i]) <= 1) {
            in_circle++;
        }
    }
    return 4 * (double)in_circle / count;
}

int main() {
    srand(time(NULL));
    for (int i = 1; i <= 100000; i++) {
        printf("%6d %f\n", i, approximate(i));
    }
    return EXIT_SUCCESS;
}

File: scala/APiRoximate.scala
import scala.util.Random

class Point(val x: Double, val y: Double) {
  def abs: Double = {
    Math.sqrt(x*x + y*y)
  }
}

object APiRoximate {
  def randomPoint(): Point = {
    val random = Random
    new Point(random.nextDouble, random.nextDouble)
  }

  def randomPoints(count: Int): Array[Point] = {
    var result: Array[Point] = new Array(count)
    for (i <- 0 until count) {
      result(i) = randomPoint()
    }
    result
  }

  def approximate(count: Int): Double = {
    val inCircle: Double = randomPoints(count).filter(_.abs <= 1).length
    4 * (inCircle / count)
  }

  def main(args: Array[String]): Unit = {
    for (n <- 1 to 100000) {
      println(s"$n ${approximate(n)}")
    }
  }
}
